### ISSUE TITLE
koreader failing on a specific cbt file

### ISSUE THREAD

**Issue by q3cpma** [koreader/koreader#5624]
* KOReader version: v2019.07
* Device: Kobo Aura HD

#### Issue
I get "no book information: not readable by engine" on this file: https://files.catbox.moe/zad3jn.cbt

Since I reproduced this issue with desktop mupdf, here's the upstream bug: https://bugs.ghostscript.com/show_bug.cgi?id=701936

**Comment by Frenzie** [comment]
Duplicate of #4287?

**Comment by q3cpma** [comment]
On Thu, Nov 21, 2019 at 02:22:26PM -0800, Frans de Jonge wrote:
> Duplicate of #4287?
Nah, this was made using GNU tar, and that previous was fixed a long time ago, anyway.

**Comment by q3cpma** [comment]
On Thu, Nov 21, 2019 at 02:22:26PM -0800, Frans de Jonge wrote:
> Duplicate of #4287?
Nah, this was made using GNU tar, and that mupdf bug was fixed a long time ago, anyway.

**Comment by NiLuJe** [comment]
We're still using that long-ago MµPDF version, though ;).

(c.f., https://github.com/koreader/koreader-base/pull/762)

**Comment by q3cpma** [comment]
I fixed it and sent a patch to upstream. Might be a good idea to update once it lands. The could use libarchive, though, it'd make everything easier.

**Comment by q3cpma** [comment]
It got fixed upstream, along with some other important tar fixes like support
for long paths (a bit late, since they advertised GNU tar support in the loader!).

**Comment by Frenzie** [comment]
Excellent! :-) Unfortunately updating MuPDF isn't super easy with the API changes.
